Samples/1a.Authentication/Program.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+
+using SteamKit2;
+
+if ( args.Length < 2 )
+{
+    Console.WriteLine( "Sample1: No username and password specified!" );
+    return;
+}
+
+// save our logon details
+var user = args[ 0 ];
+var pass = args[ 1 ];
+
+// create our steamclient instance
+var steamClient = new SteamClient();
+// create the callback manager which will route callbacks to function calls
+var manager = new CallbackManager( steamClient );
+
+// get the authentication handler, which used for authenticating with Steam
+var auth = steamClient.GetHandler<SteamAuthentication>();
+
+// get the steamuser handler, which is used for logging on after successfully connecting
+var steamUser = steamClient.GetHandler<SteamUser>();
+
+// register a few callbacks we're interested in
+// these are registered upon creation to a callback manager, which will then route the callbacks
+// to the functions specified
+manager.Subscribe<SteamClient.ConnectedCallback>( OnConnected );
+manager.Subscribe<SteamClient.DisconnectedCallback>( OnDisconnected );
+
+manager.Subscribe<SteamUser.LoggedOnCallback>( OnLoggedOn );
+manager.Subscribe<SteamUser.LoggedOffCallback>( OnLoggedOff );
+
+var isRunning = true;
+
+Console.WriteLine( "Connecting to Steam..." );
+
+// initiate the connection
+steamClient.Connect();
+
+// create our callback handling loop
+while ( isRunning )
+{
+    // in order for the callbacks to get routed, they need to be handled by the manager
+    manager.RunWaitCallbacks( TimeSpan.FromSeconds( 1 ) );
+}
+
+async void OnConnected( SteamClient.ConnectedCallback callback )
+{
+    Console.WriteLine( "Connected to Steam! Logging in '{0}'...", user );
+
+    /*
+    var authSession = await auth.BeginAuthSessionViaQR( new SteamAuthentication.AuthSessionDetails() );
+
+    Console.WriteLine( $"QR Link: {authSession.ChallengeURL}" );
+    */
+
+    // Begin authenticating via credentials
+    var authSession = await auth.BeginAuthSessionViaCredentials( new SteamAuthentication.AuthSessionDetails
+    {
+        Username = user,
+        Password = pass,
+        Persistence = SteamKit2.Internal.ESessionPersistence.k_ESessionPersistence_Ephemeral,
```

Comment:
It doesn't seem right that a fairly core logon API needs to use `SteamKit2.Internal`

Reply:
I changed it to a bool.